### PR TITLE
Allow connection between DPI and linseed

### PR DESCRIPTION
**Pull request by asincu**
## Description

- Allow connection between DPI and Linseed
- Generate certificates for DPI for mTLS connection
- Remove connection to es-gateway

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/intrusiondetection/dpi/dpi_test.go [hunk 1]
```diff
@@ -220,6 +224,7 @@ var _ = Describe("DPI rendering tests", func() {
 
 		ds := rtest.GetResource(resources, dpi.DeepPacketInspectionName, dpi.DeepPacketInspectionNamespace, "apps", "v1", "DaemonSet").(*appsv1.DaemonSet)
 		Expect(ds.Spec.Template.Spec.Containers[0].Env).Should(ContainElements(
+			// TODO: Alina - checks vars
```
Comment (caseydavenport) [inline pkg/render/intrusiondetection/dpi/dpi_test.go:227]: TODO needs to be cleaned up?

File: pkg/controller/intrusiondetection/intrusiondetection_controller.go [hunk 2]
```diff
@@ -567,7 +567,14 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
-	typhaNodeTLS.TrustedBundle.AddCertificates(esgwCertificate)
+	typhaNodeTLS.TrustedBundle.AddCertificates(linseedCertificate)
```
Comment (caseydavenport) [inline pkg/controller/intrusiondetection/intrusiondetection_controller.go:570]: I don't understand why the TyphaNodeTLS trusted bundle needs to have the Linseed certificate (or the EGW certificate) - do you?

File: pkg/render/intrusiondetection/dpi/dpi.go [hunk 3]
```diff
@@ -216,6 +215,12 @@ func (d *dpiComponent) dpiEnvVars() []corev1.EnvVar {
 		{Name: "DPI_TYPHACERTFILE", Value: d.cfg.TyphaNodeTLS.NodeSecret.VolumeMountCertificateFilePath()},
 		{Name: "DPI_TYPHAKEYFILE", Value: d.cfg.TyphaNodeTLS.NodeSecret.VolumeMountKeyFilePath()},
 		{Name: "DPI_FIPSMODEENABLED", Value: operatorv1.IsFIPSModeEnabledString(d.cfg.Installation.FIPSMode)},
+
+		{Name: "CLUSTER_NAME", Value: d.cfg.ESClusterConfig.ClusterName()},
+
+		{Name: "LINSEED_CLIENT_CERT", Value: d.cfg.DPICertSecret.VolumeMountCertificateFilePath()},
+		{Name: "LINSEED_CLIENT_KEY", Value: d.cfg.DPICertSecret.VolumeMountKeyFilePath()},
+		{Name: "FIPS_MODE_ENABLED", Value: operatorv1.IsFIPSModeEnabledString(d.cfg.Installation.FIPSMode)},
```
Comment (caseydavenport) [inline pkg/render/intrusiondetection/dpi/dpi.go:223]: Isn't this redundant with `DPI_FIPSMODEENBLED`?

File: pkg/controller/intrusiondetection/intrusiondetection_controller.go (hunk 2 above)
Comment (asincu) [reply pkg/controller/intrusiondetection/intrusiondetection_controller.go:570]: DeepPacketInspection queries Typha to get updates for resources like new workload endpoints. Most likely, the esgateway cert was bundled up together.

**Review by caseydavenport — APPROVED** [review]
Two questions / minor comments but otherwise LGTM

File: pkg/render/intrusiondetection/dpi/dpi.go (hunk 3 above)
Comment (asincu) [reply pkg/render/intrusiondetection/dpi/dpi.go:223]: Yup, I kinda find it simpler to have FIPS_MODE_ENABLED everywhere. I will make the change in dpi.

Comment (caseydavenport) [reply pkg/render/intrusiondetection/dpi/dpi.go:223]: We should remove setting DPI_FIPSMODEENABLED in this PR then instead of having both env vars set

**Pull request merged**